blockfetcher/tracer_fetcher.go

```diff
@@ -0,0 +1,107 @@
+package blockfetcher
+
+import (
+	"context"
+	"encoding/base64"
+	"fmt"
+	pbbstream "github.com/streamingfast/bstream/pb/sf/bstream/v1"
+	"github.com/streamingfast/eth-go/rpc"
+	"github.com/streamingfast/firehose-ethereum/block"
+	pbeth "github.com/streamingfast/firehose-ethereum/types/pb/sf/ethereum/type/v2"
+	"go.uber.org/zap"
+	"google.golang.org/protobuf/proto"
+	"google.golang.org/protobuf/types/known/anypb"
+	"google.golang.org/protobuf/types/known/timestamppb"
+	"strings"
+	"time"
+)
+
+type TracerBlockFetcher struct {
+	fetcher *BlockFetcher
+}
+
+func (f *TracerBlockFetcher) IsBlockAvailable(requested uint64) bool {
+	return f.fetcher.IsBlockAvailable(requested)
+}
+
+func (f *TracerBlockFetcher) Fetch(ctx context.Context, client *rpc.Client, blkNum uint64) (b *pbbstream.Block, skipped bool, err error) {
+	respStr, err := client.DoRequest(ctx, "debug_traceFirehoseBlockByNumber", []interface{}{fmt.Sprintf("0x%x", blkNum), nil})
+	if err != nil {
+		return nil, false, fmt.Errorf("failed to trace block %d: %w", blkNum, err)
+	}
+
+	// Remove surrounding quotes
+	if len(respStr) > 1 && respStr[0] == '"' && respStr[len(respStr)-1] == '"' {
+		respStr = respStr[1 : len(respStr)-1]
+	}
```

Comment:
We should actually exchange right away the correct information, probably `pbpstream.Block`, this way, you would avoid all the work being done here while it could have been done on the server directly which would make the interface easier.

Essentially, it's about moving all the logic here inside the `debug_traceFirehoseBlockByNumber` server handler.